### PR TITLE
🚀 feat(cyclops-ui): Copy Module Name and Namespace buttons

### LINKED ISSUE #639: Copy Module Name and Namespace buttons
**Describe the solution you'd like**
Implement two buttons. The first should be placed next to the module name and the second to the module namespace (check out the orange Xs in the image below).

The first button copies the module's name to your clipboard, while the second copies the namespace. Use [AntDesign Icons](https://ant.design/components/icon)).

Screenshot 2024-10-18 at 12 56 13

### PR DESCRIPTION
**Pull request by ArnavK-09**
closes #639

## 📑 Description

image

## ✅ Checks
- [x] I have tested my code (provide screenshots or screen recordings of a working solution)
- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by ArnavK-09** [comment]
@quest-bot loot #639

File: cyclops-ui/src/components/pages/ModuleDetails.tsx [hunk 1]
```diff
@@ -784,6 +802,24 @@ const ModuleDetails = () => {
...
+                onClick={() => {
+                  navigator.clipboard.writeText(module.targetNamespace);
+                }}
+                style={{
+                  marginLeft: "4px",
+                  padding: "2px 1.5px",
+                }}
+              >
+                <CopyOutlined
+                  style={{
```
Comment (KaradzaJuraj) [inline cyclops-ui/src/components/pages/ModuleDetails.tsx:817]: Color the icon grey

File: cyclops-ui/src/components/pages/ModuleDetails.tsx [hunk 2]
```diff
@@ -752,6 +752,24 @@ const ModuleDetails = () => {
...
+                  type="text"
+                  onClick={() => {
+                    navigator.clipboard.writeText(module.name);
+                  }}
+                  style={{
+                    marginLeft: "4px",
+                    padding: "2px",
+                  }}
+                >
+                  <CopyOutlined
```
Comment (KaradzaJuraj) [inline cyclops-ui/src/components/pages/ModuleDetails.tsx:766]: Color the icon grey

**Review by KaradzaJuraj — CHANGES_REQUESTED** [review]
Hey @ArnavK-09, the icons are currently positioned at the bottom, but I'd like them to be positioned at the top.

You will probably need to change the styling of the `<Title>` to use `flex` for the module name copy. For the namespace copy, you can update the styling of the `Description.Item`

Also, color the icons grey!

**Comment by ArnavK-09** [comment]
> Hey @ArnavK-09, the icons are currently positioned at the bottom, but I'd like them to be positioned at the top.
> 
> You will probably need to change the styling of the `<Title>` to use `flex` for the module name copy. For the namespace copy, you can update the styling of the `Description.Item`
> 
> Also, color the icons grey!

"top"

image

**Review by KaradzaJuraj — CHANGES_REQUESTED** [review]
Hey @ArnavK-09 , sorry for doing this but I dislike this solution (my fault, I proposed it).

I think a better solution would be if just by clicking on the module name (or namespace), it copies it to your clipboard.

I propose the following:
1) Remove the added buttons and styling
2) Wrap `{module.name}` & `{module.targetNamespace}` in `<span>` with the `OnClick` effect of copying + style the cursor when hovering to "pointer"
3) Wrap the new `<span>` with the `<Tooltip>` component so that when people hover over the name/namespace, a tooltip will appear to tell them they can copy it
4) update the tooltip text to "Copy module name to clipboard" and "Copy namespace to clipboard"

Let me know if this makes sense.

**Comment by ArnavK-09** [comment]
> Hey @ArnavK-09 , sorry for doing this but I dislike this solution (my fault, I proposed it).
> 
> I think a better solution would be if just by clicking on the module name (or namespace), it copies it to your clipboard.
> 
> I propose the following:
> 1) Remove the added buttons and styling
> 2) Wrap `{module.name}` & `{module.targetNamespace}` in `<span>` with the `OnClick` effect of copying + style the cursor when hovering to "pointer"
> 3) Wrap the new `<span>` with the `<Tooltip>` component so that when people hover over the name/namespace, a tooltip will appear to tell them they can copy it
> 4) update the tooltip text to "Copy module name to clipboard" and "Copy namespace to clipboard"
> 
> Let me know if this makes sense.

Yup, this looks better, 
I'll implement this asap and resolve this issue 
Thanks

**Comment by ArnavK-09** [comment]
> Hey @ArnavK-09 , sorry for doing this but I dislike this solution (my fault, I proposed it).
> 
> I think a better solution would be if just by clicking on the module name (or namespace), it copies it to your clipboard.
> 
> I propose the following:
> 1) Remove the added buttons and styling
> 2) Wrap `{module.name}` & `{module.targetNamespace}` in `<span>` with the `OnClick` effect of copying + style the cursor when hovering to "pointer"
> 3) Wrap the new `<span>` with the `<Tooltip>` component so that when people hover over the name/namespace, a tooltip will appear to tell them they can copy it
> 4) update the tooltip text to "Copy module name to clipboard" and "Copy namespace to clipboard"
> 
> Let me know if this makes sense.

Like this?

**Comment by ArnavK-09** [comment]
> Just a small change to finish this off 😅

sure, is this alr?

**Review by KaradzaJuraj — APPROVED** [review]
Looking good! Thanks @ArnavK-09 🙏

**Pull request merged**